### PR TITLE
Pre Emulator Launch Script

### LINKED ISSUE #246: How to run a script before launching the emulator?
## Description
I need to insert an image into the Android emulator camera (https://stackoverflow.com/a/64922184/8358501). For this I need to change some configs files of the emulator **before** launching it. 

Is there an option to do so? 

If not, could we add a new optional parameter for the action, like `script-before-launching` which acts like `script`, but runs before launching the emulator?

### PR DESCRIPTION
**Pull request by nilsreichardt**
## Description
I added a new parameter `pre-emulator-launcher-script` (if you have better names, please let me know). With this parameter, you can pass a custom script which will be executed before launching the emulator (see #246 for more context).

## Tests
I haven't written any unit tests, but added the `pre-emulator-launcher-script` to the CI of 1 job, so that we have 1 job with the parameter and 1 without the parameter. You can see here the ouput: https://github.com/nilsreichardt/android-emulator-runner/runs/6196378411?check_suite_focus=true#step:9:97

## Related Tickets
Closes #246

### CODE REVIEW

**Comment by mrk-han** [comment]
Would this same behavior be possible by separating the command to create the emulator, and the command to run the tests with a `run:` step? 

Also, this seems less efficient than making a post-create-avd script to edit the config after the AVD is created. If you do this once, post-create, then you won't need to run it again if the emulator is cached. Post-create script would be able to be run in a emulator caching step.

**Comment by mrk-han** [comment]
@nilsreichardt I suppose I do see the benefit of being able to call a script if using one instance of this action in your .yml 

@ychescale9 Can we merge this in?

File: README.md [hunk 1]
```diff
@@ -155,11 +155,12 @@ jobs:
...
 | `emulator-build` | Optional | N/A | Build number of a specific version of the emulator binary to use e.g. `6061023` for emulator v29.3.0.0. |
-| `working-directory` | Optional | `./` | A custom working directory - e.g. `./android` if your root Gradle project is under the `./android` sub-directory within your repository. |
+| `working-directory` | Optional | `./` | A custom working directory - e.g. `./android` if your root Gradle project is under the `./android` sub-directory within your repository. Will be used for `script` & `pre-emulator-launch-script`. |
 | `ndk` | Optional | N/A | Version of NDK to install - e.g. `21.0.6113669` |
 | `cmake` | Optional | N/A | Version of CMake to install - e.g. `3.10.2.4988404` |
 | `channel` | Optional | stable | Channel to download the SDK components from - `stable`, `beta`, `dev`, `canary` |
 | `script` | Required | N/A | Custom script to run - e.g. to run Android instrumented tests on the emulator: `./gradlew connectedCheck` |
+| `pre-emulator-launch-script` | Optional | N/A | Custom script to run which will run before launch the emulator - e.g. `./adjust-emulator-configs.sh` |
```
Comment (ychescale9) [inline README.md:163]:
```suggestion
| `pre-emulator-launch-script` | Optional | N/A | Custom script to run after creating the AVD and before launching the emulator - e.g. `./adjust-emulator-configs.sh` |
```

**Comment by mrk-han** [comment]
```
Exception in thread "main" java.lang.IllegalStateException: The same set of inputs should exist in action manifest and types manifest!
[4](https://github.com/ReactiveCircus/android-emulator-runner/actions/runs/3302783629/jobs/5450189503#step:4:5)
	at it.krzeminski.githubactionstyping.ManifestsToReportKt.manifestsToReport(ManifestsToReport.kt:22)
[5](https://github.com/ReactiveCircus/android-emulator-runner/actions/runs/3302783629/jobs/5450189503#step:4:6)
	at it.krzeminski.githubactionstyping.MainKt.main(Main.kt:19)
[6](https://github.com/ReactiveCircus/android-emulator-runner/actions/runs/3302783629/jobs/5450189503#step:4:7)
	at it.krzeminski.githubactionstyping.MainKt.main(Main.kt)
```
Just need to address this failed check

**Comment by nilsreichardt** [comment]
> Just need to address this failed check

Is fixed :)

**Comment by mrk-han** [comment]
Some of these emulators have been so flakey lately when booting, and I don't think we have a way of rerunning failed runs unless @ychescale9 kicks them off. I'm trying to address it in my PR with a mechanism to comment "run tests" on a PR.

Failed job: https://github.com/ReactiveCircus/android-emulator-runner/actions/runs/3328847468/jobs/5506688370

**Pull request merged**